header-rewrite-filter/header_processor.cc

```diff
@@ -412,15 +423,20 @@ namespace HeaderRewriteFilter {
     return Utility::StringToFunctionType(dynamic_function);
   }
 
-  absl::Status DynamicFunctionProcessor::parseOperation(absl::string_view function_expression, const bool isRequest) {
-    function_type_ = getFunctionType(function_expression);
+  absl::Status DynamicFunctionProcessor::parseOperation(absl::string_view function_expression) {
+    // make sure dynamic function is wrapped in %[]
+    if (function_expression.substr(0, 2) != Utility::DYNAMIC_FUNCTION_DELIMITER.substr(0, 2) && function_expression.substr(function_expression.size()-1, 1) != Utility::DYNAMIC_FUNCTION_DELIMITER.substr(2, 1)) {
```

Comment:
Does something before this call guarantee `function_expression` is at least len 2? If not, this needs to check that to avoid crashing on bad input

(it wouldn't be bad to add an extra safety and check it anyway)